### PR TITLE
doc: Moved 54H doc folder

### PR DESCRIPTION
**Pull request by FrancescoSer**
Moved 54H doc folder
Updated redirects.py

### CODE REVIEW

File: doc/_utils/redirects.py [hunk 1]
```diff
@@ -203,31 +203,6 @@
     ("app_dev/device_guides/working_with_nrf/nrf54l/nrf54l15_gs", "gsg_guides"),
     ("device_guides/working_with_nrf/nrf54l/testing_dfu", "app_dev/device_guides/nrf54l/testing_dfu"), # Testing the DFU solution
     ("app_dev/device_guides/working_with_nrf/nrf54l/testing_dfu", "app_dev/device_guides/nrf54l/testing_dfu"),
-    ("device_guides/nrf54h", "app_dev/device_guides/nrf54h"), # Developing with nRF54H Series
```
Comment (divipillai) [inline doc/_utils/redirects.py:206]: The deleted entries should be kept and should re-direct to the latest url. Also a new entry is needed for the existing path. For example, for the ug_nrf54h20_architecture, we will have two entries:

* ("device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture", "app_dev/device_guides/nrf54h/ug_nrf54h20_architecture"), # Architecture of nRF54H20 (landing)
*  "app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture", "app_dev/device_guides/nrf54h/ug_nrf54h20_architecture"), # Architecture of nRF54H20 (landing)

Same applies for all the entries.

File: doc/_utils/redirects.py [hunk 2]
```diff
@@ -203,31 +203,6 @@
     ("app_dev/device_guides/working_with_nrf/nrf54l/nrf54l15_gs", "gsg_guides"),
```
Comment (divipillai) [inline doc/_utils/redirects.py:203]: Also, tag file would require an update - nrf\doc\_zoomin\ncs.tags

File: doc/_utils/redirects.py (hunk 1 above)
Comment (peknis) [reply doc/_utils/redirects.py:206]: I was also a bit suspicious with the redirects. I believe Divya is right about this.

File: doc/_utils/redirects.py (hunk 2 above)
Comment (umapraseeda) [reply doc/_utils/redirects.py:203]: Yes, the line here needs to be updated with the correct path: https://github.com/nrfconnect/sdk-nrf/blob/main/doc/_zoomin/ncs.tags.yml#L209

File: CODEOWNERS [hunk 3]
```diff
@@ -83,8 +83,8 @@
 /doc/nrf/app_dev/device_guides/nrf91/     @nrfconnect/ncs-modem-doc
 /doc/nrf/app_dev/device_guides/nrf91/nrf91_snippet.rst @nrfconnect/ncs-cia-doc
 /doc/nrf/app_dev/device_guides/pmic/      @nrfconnect/ncs-pmic-doc
-/doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ @FrancescoSer
-/doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_*.rst @nrfconnect/ncs-charon-doc
+/doc/nrf/app_dev/device_guides/nrf54h/ @FrancescoSer
+/doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_*.rst @nrfconnect/ncs-charon-doc
```
Comment (greg-fer) [inline CODEOWNERS:87]: Is renaming `Working with nRF54H20` -> `Developing with nRF54H20` planned for this PR/task or a following one?

Comment (FrancescoSer) [reply CODEOWNERS:87]: The renaming could take place, but it is my understanding that that renaming was connected to the availability of the quick start for the SoC with the precompiled binaries, which unfortunately are not ready yet.

Comment (greg-fer) [reply CODEOWNERS:87]: Renaming of the section name has nothing to do with the renaming and replacement of the getting started guide.

**Review by greg-fer — APPROVED** [review]

**Review by maciejpietras — APPROVED** [review]

**Comment by FrancescoSer** [comment]
@umapraseeda @b-gent The PR is ready, please rereview

File: doc/_utils/redirects.py (hunk 2 above)
Comment (FrancescoSer) [reply doc/_utils/redirects.py:203]: updated

**Review by b-gent — APPROVED** [review]

**Review by divipillai — APPROVED** [review]
Approved. Just a minor comment.

**Review by umapraseeda — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**